### PR TITLE
FAS-3 cleanup and cosmetic changes

### PR DESCRIPTION
**Pull request by dakq**

### CODE REVIEW

File: fastqheat.py [hunk 1]
```diff
@@ -10,131 +10,106 @@
...
-    SRR_pattern = re.compile(r'^(SRR|ERR|DRR)\d+$')
-    SRP_pattern = re.compile(
-        r'^(((SR|ER|DR)[PAXS])|(SAM(N|EA|D))|PRJ(NA|EB|DB)|(GS[EM]))\d+$'
-    )
-    if method == "f":
-        if SRR_pattern.search(term) is not None:
-            accession = term
+
+def _handle_f_method(term, out):
+    if SRR_PATTERN.search(term) is not None:
```
Comment (RandyRomero) [inline fastqheat.py:20]: `if SRR_PATTERN.search(term):`

**Review by RandyRomero — CHANGES_REQUESTED** [review]

File: download.py [hunk 2]
```diff
@@ -76,21 +76,22 @@ def download_run_ftp(accession, term, out):
...
         # execute command in commandline
         os.system(bash_command)
         # check completeness of the file and return boolean
         correctness.append(md5_checksum(SRR, f"{out}/{term}", md5))
+
     if all(correctness):
         logging.info("Current Run: %s has been successfully downloaded", accession)
-    return all(correctness)
+        return True
+    else:
```
[the remark this answers is not in the document]
Comment (dakq) [reply download.py:93]: No, but I prefer this version to

    if all(correctness):
        logging.info("Current Run: %s has been successfully downloaded", accession)
        return True
    return False

File: fastqheat.py [hunk 3]
```diff
@@ -10,131 +10,106 @@
 from download import download_run_aspc, download_run_fasterq_dump, download_run_ftp
 from study_info.get_sra_study_info import get_run_uid
 
+SRR_PATTERN = re.compile(r'^(SRR|ERR|DRR)\d+$')
```
[the remark this answers is not in the document]
Comment (dakq) [reply fastqheat.py:13]: I disagree, because those constants aren't used anywhere else in the project.

File: fastqheat.py (hunk 1 above)
Comment (dakq) [reply fastqheat.py:20]: Functionally equivalent, strongly depends on personal preference. I prefer comparing values with `None` explicitly.

File: download.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dakq) [reply download.py]: Removed `else` clauses.

File: fastqheat.py (hunk 1 above)
Comment (dakq) [reply fastqheat.py]: Removed explicit `is not None` comparisons.

**Review by RandyRomero — APPROVED** [review]
:+1:

File: fastqheat.py (hunk 1 above)
Comment (dakq) [reply fastqheat.py]: > if search() returns None or any other nullable

`re.search` is documented to  "Return None if no position in the string matches the pattern". It cannot return any other falsy value.

Comment (RandyRomero) [reply fastqheat.py]: So if it cannot return any other nullable values, why check whether it is `None` and not "", [], {}, 0?

If we had code like these:
```python
if foo is None:
    do_this()
elif foo == []:
    do_that()
```
that would make sense

Comment (dakq) [reply fastqheat.py]: Like I said, personal preference.

Comment (RandyRomero) [reply fastqheat.py]: I don't quite understand how it is about personal preferences, to be honest. If it is not necessary to check exactly for None, why check exactly for None? There should be a reason to do or not to do something, imo. We code for money, we refactor code to make it more maintainable, we check for `None` if it makes any difference. If not, why?

**Pull request merged**